File: backend/Core/Entities/Record.cs
using SiradigCalc.Core.Entities.Base;

namespace SiradigCalc.Core.Entities;

public class Record : Entity
{
    public Guid TemplateId { get; set; } = default!;
    public RecordTemplate Template { get; set; } = null!;
    public string Title { get; set; } = string.Empty;
    public ICollection<RecordValue> Values { get; set; } = [];
    public ICollection<RecordConversion> ConvertedTo { get; set; } = [];
    public ICollection<RecordConversion> ConvertedFrom { get; set; } = [];
}

File: frontend/app/data/interfaces/Record.ts
import { Template } from "@/data/interfaces/Template";
import { RecordValue } from "@/data/interfaces/RecordValue";

export interface Record {
  templateId: string;
  template: Template;
  title: string;
  values: Array<RecordValue>;
  convertedTo: Array<any>;
  convertedFrom: Array<any>;
  id: string;
  createdAt: Date;
  updatedAt: Date;
  deleted: boolean;
}

export interface RecordFlat {
  id: string;
  title: string;
  name: string;
  description: string;
  recordTemplateId: string;
  values: Array<RecordValue>;
}
